frontend/catalyst/compiled_functions.py

```
@@ -160,6 +161,8 @@ def _exec(shared_object, has_return, out_type, numpy_dict, *args):
         Returns:
             the return values computed by the function or None if the function has no results
         """
+        if shared_object is None:
+            return None
```

I feel like calling `open` should not silently fail 🤔 If you call open you probably expect the library to be actually loaded.

---

So what I did, was to skip the `open` if `shared_object_file` is null which results in `shared_object` to be null

---

Oh sorry, I was reading this wrong thinking it was in the `open` function 🤦‍♂️

---

I just noticed that we don't actually need this line, since if `lower_to_llvm` is False, we will never attempt to execute the compiled function.